### PR TITLE
Adding Other Shelves

### PR DESCRIPTION
**Pull request by jmwright**
There are shelf assemblies outlined in the database and in the code that are not accounted for yet. This PR is an attempt to add some of those. #94 needs to be merged first, and potentially the new CadOrchestrator changes need to be merged and then used to reshape this PR.

### CODE REVIEW

**Comment by jmwright** [comment]
Examples of other shelf assemblies that are being added. There is an automatically captured image of the assembly, followed by an image of the exploded assembly, for each shelf.
assembly_render_test_Hex
assembly_render_test_exploded_Hex
assembly_render_test_NUC8I5BEH
assembly_render_test_exploded_NUC8I5BEH
assembly_render_test_Raspberry Pi 4B
assembly_render_test_exploded_Raspberry Pi 4B
assembly_render_test_Unifi Flex Mini
assembly_render_test_exploded_Unifi Flex Mini
assembly_render_test_Unifi Switch Flex
assembly_render_test_exploded_Unifi Switch Flex
assembly_render_test_Western Digital Blue SSD
assembly_render_test_exploded_Western Digital Blue SSD
assembly_render_test_Western Digital Red HDD
assembly_render_test_exploded_Western Digital Red HDD

**Comment by julianstirling** [comment]
Awesome, we are really cooking on gas now. I'm going to try to get this all synced up so the renders are generated from the CadOrchestrator commands and server

File: nimble_build_system/cad/shelf.py
Comment (julianstirling) [inline nimble_build_system/cad/shelf.py:418]: Probably in future when disabling it would be best to disable by name 

i.e. `disable=too-many-statements` rather than `disable=R0915` as it is more human readable.

File: nimble_build_system/cad/fasteners.py
Comment (julianstirling) [inline nimble_build_system/cad/fasteners.py:158]: Is there a reason that the parameter is `axis` in `Ziptie` but `directional_axis` in the parent class `Fastener`?

**Review by julianstirling — APPROVED** [review]
Hi @jmwright 

Great work here I have checked over the code and made a couple of comments and a minor suggestion.

I realise however than on rebasing I have slightly buggered up the renders, probably to do with the fastener axis:

assembly_render_test_exploded_Hex
assembly_render_test_exploded_Unifi Flex Mini
assembly_render_test_exploded_Western Digital Red HDD

Is it OK if I leave this to you to fix? I am going to click approve on this so you can merge once you fix my mistake

File: nimble_build_system/cad/fasteners.py
Comment (jmwright) [reply nimble_build_system/cad/fasteners.py:156]: I think I was trying to keep the variables separate as they're passed through the constructor and super calls on `Screw` and `Ziptie`.

**Comment by jmwright** [comment]
@julianstirling I don't think you messed the alignment up. Please make sure that cq-annotate is up-to-date in your environment with the latest in the main branch on git.

**Pull request merged**
